DSD-2150: fixing broken image references

bigfishdesign13 opened the pull request:
Fixes JIRA ticket [DSD-2150](https://newyorkpubliclibrary.atlassian.net/browse/DSD-2150)

## This PR does the following:

- Fixed the image references in the `Using Next.js Image` Development Guide and in the `Responsive Grid` Style Guide.

NOTE: The syntax of the image references was updated to mirror what was being used in the `Filtering` Style Guide.

## How has this been tested?

<!-- Please describe in detail how you tested your changes. -->

- local Storybook

## Accessibility concerns or updates

<!-- Describe any accessibility concerns or updates that were made that should be known. -->

-

### Accessibility Checklist

- [ ] Checked Storybook's "Accessibility" tab for color contrast and other issues.
- [ ] The feature works with keyboard inputs including tabbing back and forward and pressing space, enter, arrow, and esc keys.
- [ ] For hidden text or when `aria-live` is used, a screenreader was used to verify the text is read.
- [ ] For features that involve UI updates and focusing on DOM `ref`s, focus management was reviewed.
- [ ] The feature works when the page is zoomed in to 200% and 400%.

## Open Questions

<!-- Any questions you want to ask the reviewer? -->

-

### Checklist:

<!-- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!-- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->

- [ ] I have updated the Storybook documentation and changelog accordingly.
- [ ] I have added relevant accessibility documentation for this pull request.
- [ ] All new and existing tests passed.

### Front End Review:

<!-- This step is done AFTER creating a PR. -->
<!-- Vercel creates a static Storybook preview URL once the PR is created. -->
<!-- That preview URL is added by Vercel as a comment. -->

- [ ] Review the Vercel preview deployment once it is ready.


[DSD-2150]: https://newyorkpubliclibrary.atlassian.net/browse/DSD-2150?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

bigfishdesign13 committed:
fixed image references

EdwinGuzman commented on src/components/DevelopmentGuide/UsingNextjsImage.mdx:
```diff
@@ -476,4 +477,4 @@ In the example above:
 - If viewport width ≤ `480px` → image uses `100%` of viewport width.
 - If viewport width ≤ `1024px` → image uses `50%` of viewport width.
-- Otherwise → image uses `25%` of viewport width.
+- Otherwise → image uses `25%` of viewport w{nextImageFallbackLight}
```
typo, but also, this seems like a mistake

bigfishdesign13 replied:
Yes! I move too fast in VS Code sometimes and I click before letting go of the track pad. Then multiple lines of code are selected and whatever I enter is added in both places.

EdwinGuzman reviewed (commented):
I don't think that adding `/public/` is going to fix the issue in the static build:

<img width="537" height="288" alt="Image" src="https://github.com/user-attachments/assets/12bcae19-ea70-4bd5-99cb-936998fd4293" />

The images should be relative:

<img width="504" height="273" alt="Image" src="https://github.com/user-attachments/assets/ab653783-df92-4960-a7fc-637192671cb7" />

I haven't tested this locally, so I'm not sure if it'll work in the static build but not locally.

-----

Conventionally, I'm not in favor of importing images as js variables. It _feels_ wrong and although next.js supports it, it's not a pattern that's used elsewhere. Perhaps not great, but I'd revert back to inline images unless you're strongly against it.

EdwinGuzman commented:
I also missed that this was done in the filtering docs. This follows the same `/public/` pattern, but taking a look at the source:
<img width="961" height="48" alt="Screenshot 2025-09-11 at 10 22 28 AM" src="https://github.com/user-attachments/assets/e0416839-cbfa-48af-8b3e-335c3bebf385" />

I don't think it's a big deal, but just pointing out it's not relative image requests.

Edit: we can try setting up the static directory: https://storybook.js.org/docs/configure/integration/images-and-assets

bigfishdesign13 committed:
fixed typo

bigfishdesign13 committed:
using Storybook congif to implement static files

bigfishdesign13 committed:
one more code cleanup

bigfishdesign13 commented:
@EdwinGuzman I reverted to the format that was previously used, but I also updated the Storybook config to add `staticDirs` references. But now I am not sure how to properly test the change to see if it works. Right now, everything looks fine on the DS dev site and that is running the same code as what is on the DS prod site, where the images are broken.

bigfishdesign13 committed:
Merge branch 'development' into DSD-2150/docs-image-references

bigfishdesign13 committed:
alphabetized objects in Storybook config

bigfishdesign13 committed:
applied same syntax to other imstances where local images are used

EdwinGuzman reviewed (approved):
Let's test it in the next release.

bigfishdesign13 committed:
Merge branch 'development' into DSD-2150/docs-image-references
